apps/bfd-pipeline/bfd-pipeline-rda-grpc/src/main/java/gov/cms/bfd/pipeline/rda/grpc/source/GrpcRdaSource.java

```diff
@@ -333,6 +343,29 @@ private ManagedChannel createChannel() {
     public CallOptions createCallOptions() {
       CallOptions answer = CallOptions.DEFAULT;
       if (authenticationToken != null) {
+        /**
+         * The RDA API uses a JWT token for authentication, by design this is set to expire X days
+         * after being issued. This check will alert the team of a token that is close to expiring
+         * so we can coordinate having a new one issued by the RDA API team before authentication
+         * fails.
+         */
+        if (expirationDate != null) {
+          long daysToExpire =
+              Instant.now().until(Instant.ofEpochMilli(expirationDate * 1000), ChronoUnit.DAYS);
+
+          if (daysToExpire < 0) {
+            LOGGER.error("JWT is expired!");
+          } else if (daysToExpire < 31) {
+            String logMessage = String.format("JWT will expire in %d days", daysToExpire);
+
+            if (daysToExpire < 14) {
+              LOGGER.error(logMessage);
+            } else {
+              LOGGER.warn(logMessage);
+            }
```

Comment:
I guess the critical thing here is that these logs need alerts configured for all errors and warnings so that this goes somewhere that somebody will see it.

Reply:
For the error, I believe it will work with this alarm https://github.com/CMSgov/beneficiary-fhir-data/blob/695d47ab8f2ffadbf25efe17bcdb55a9d31beeff/ops/terraform/modules/resources/bfd_pipeline/main.tf#L37-L48.

The warning statement though will not trigger any alarms.  I'd like to talk to you about this in the future, as I think we could enable alarms on warning statements if we did a little grooming.  Kyle and I noticed on Monday that more than 95% of the warning statements are coming from the QueryLoggingListener, with a couple of changes there I think the signal-to-noise ratio would vastly improve.